components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -1,110 +1,627 @@
-    });
+  // prescribed volcanic aerosols.
+  ssa_cmip6_sw_ =
+      mam_coupling::view_3d("ssa_cmip6_sw", ncol_, nlev_, nswbands_);
+  af_cmip6_sw_ = mam_coupling::view_3d("af_cmip6_sw", ncol_, nlev_, nswbands_);
+  ext_cmip6_sw_ =
+      mam_coupling::view_3d("ext_cmip6_sw", ncol_, nswbands_, nlev_);
+  ext_cmip6_lw_ =
+      mam_coupling::view_3d("ext_cmip6_lw_", ncol_, nlev_, nlwbands_);
+  // FIXME: We need to get ssa_cmip6_sw_, af_cmip6_sw_, ext_cmip6_sw_,
+  // ext_cmip6_lw_ from a netcdf file.
+  // We will rely on eamxx to interpolate/map data from netcdf files.
+  // The io interface in eamxx is being upgraded.
+  // Thus, I will wait until changes in the eamxx's side are completed.
+  Kokkos::deep_copy(ssa_cmip6_sw_, 0.0);
+  Kokkos::deep_copy(af_cmip6_sw_, 0.0);
+  Kokkos::deep_copy(ext_cmip6_sw_, 0.0);
+  Kokkos::deep_copy(ext_cmip6_lw_, 0.0);
+
+  // set up our preprocess/postprocess functors
+  preprocess_.initialize(ncol_, nlev_, wet_atm_, wet_aero_, dry_atm_,
+                         dry_aero_);
+  postprocess_.initialize(ncol_, nlev_, wet_atm_, wet_aero_, dry_atm_,
+                          dry_aero_);
+
+  const int work_len = mam4::modal_aer_opt::get_work_len_aerosol_optics();
+  work_              = mam_coupling::view_2d("work", ncol_, work_len);
+
+
+  // read table info
+  {
+    using namespace ShortFieldTagsNames;
+
+    using view_1d_host = typename KT::view_1d<Real>::HostMirror;
+
+    // Note: these functions do not set values for aerosol_optics_device_data_.
+    mam4::modal_aer_opt::set_complex_views_modal_aero(
+        aerosol_optics_device_data_);
+    mam4::modal_aer_opt::set_aerosol_optics_data_for_modal_aero_sw_views(
+        aerosol_optics_device_data_);
+    mam4::modal_aer_opt::set_aerosol_optics_data_for_modal_aero_lw_views(
+        aerosol_optics_device_data_);
+
+    mam_coupling::AerosolOpticsHostData aerosol_optics_host_data;
+
+    std::map<std::string, FieldLayout> layouts;
+    std::map<std::string, view_1d_host> host_views;
+    ekat::ParameterList rrtmg_params;
+
+    mam_coupling::set_parameters_table(aerosol_optics_host_data, rrtmg_params,
+                                       layouts, host_views);
+
+    std::string mam4_mode1_physical_properties_file =
+      m_params.get<std::string>("mam4_mode1_physical_properties_file");
+
+    std::string mam4_mode2_physical_properties_file =
+      m_params.get<std::string>("mam4_mode2_physical_properties_file");
+
+    std::string mam4_mode3_physical_properties_file =
+      m_params.get<std::string>("mam4_mode3_physical_properties_file");
+
+    std::string mam4_mode4_physical_properties_file =
+      m_params.get<std::string>("mam4_mode4_physical_properties_file");
+
+
+    std::vector<std::string> name_table_modes = {
+      mam4_mode1_physical_properties_file, mam4_mode2_physical_properties_file,
+      mam4_mode3_physical_properties_file, mam4_mode4_physical_properties_file};
+
+    for(int imode = 0; imode < ntot_amode; imode++) {
```

Comment:
Style preference, but throwing it out there. Rather than creating all those strings and wrap them in a vector, only to access them one by one, have you thought about just doing
```c++
for (int imode ...) {
  const auto  key = "mam4_mode" + std::to_string(imode) + "_physical_properties_file";
  const auto& fname = m_params.get<std::string>(key);
  mam_coupling::read_rrtmg_table(fname,...);
}
```
?

Reply:
Thanks. That change removed a few lines of code.